cns/ipampool/monitor.go

```diff
@@ -343,6 +347,20 @@ func (pm *Monitor) GetStateSnapshot() cns.IpamPoolMonitorStateSnapshot {
 	}
 }
 
+// GenerateARMID uses the Subnet ARM ID format to populate the ARM ID with the metadata.
+// If either of the metadata attributes are empty, then the ARM ID will be an empty string.
+func GenerateARMID(nnc *v1alpha.NodeNetworkConfig) string {
+	subscription := nnc.Status.NetworkContainers[0].SubscriptionID
+	resourceGroup := nnc.Status.NetworkContainers[0].ResourceGroupID
+	vnetID := nnc.Status.NetworkContainers[0].VNETID
+	subnetID := nnc.Status.NetworkContainers[0].SubnetID
```

Comment:
this should probably accept an NC, not the whole NNC, and the caller should pick which NC is passed in so that we have the option to do the right thing still if the NNC has multiple NCs